### ISSUE TITLE
Unresponsive Docker daemon and corrupted filesystem causing builds and deploys to fail

### ISSUE THREAD

**Issue by nzoschke** [convox/rack#314]
## WIP: 

https://github.com/convox/rack/pull/315
https://github.com/convox/agent/pull/12

## Observations

I observed deployments that get stuck due to tasks that are stuck in pending on an instance. I have also observed builds failing with an error like `OSError: [Errno 5] Input/output error: '/tmp/tmpsl27xkc4pip-wheel-'`.

In both cases, introspecting the instance shows a hung `dockerd` and filesystem errors in `dmesg`.

## Steps To Reproduce

Deploy an app with a large Docker image that exits or crashes repeatedly. Eventually the disks will fill up due to the numerous exited containers, and Docker and the filesystem will go into a bad state.

## Manual Detection and Remediation

Terminate the bad instance and wait a few minutes. The Auto Scale Group will boot a new instance. ECS will re-schedule the containers.

```
$ convox instances
ID          AGENT  STATUS  STARTED     PS  CPU    MEM   
i-1463129d  on     active  3 days ago  2   0.00%  25.56%
i-29f226a9  on     active  2 days ago  1   0.00%  19.17%
i-16ab80a5  on     active  3 days ago  0   0.00%  0.00%

$ convox api get /instances | jq '.[].id' | xargs -L1 -I% convox instances ssh % docker ps
[one of them hangs]

$ convox api get /instances | jq '.[].id' | xargs -L1 -I% convox instances ssh % dmesg
[one of them has scary output about high water marks and remounting filesystems read-only]

$ convox instances terminate  i-1463129d
```

## Engineering Playbook

- [x] Document Issue
  - [x] Observations
  - [x] Steps to reproduce
  - [x] Manual procedure to work around
- [x] Add Visibility
  - [x] Log when issue occurs
  - [x] Log metrics around issue
  - [x] Notify when issue occurs
- [x] Automate work around
  - [x] Log when automation occurs
  - [x] Log metrics around automation
- [x] Remediate root cause
- [x] Verify fix against metrics around issue and automation

## Release Playbook
- [x] Release branch (20160202173440-unresponsive-docker)
- [x] Pass CI (https://circleci.com/gh/convox/rack/390)
- [x] Merge into master
- [x] Release master (20160202191211)
- [x] Pass CI
- [x] Update dev and testing racks
- [x] Publish release (20160204174227)

**Comment by nzoschke** [comment]
Occurrence is down to zero:

<img width="782" alt="screen shot 2016-02-05 at 3 09 07 pm" src="https://cloud.githubusercontent.com/assets/147410/12861747/73845d32-cc1a-11e5-8133-0621df85eff7.png">
